lib/services/access_list.go

```diff
@@ -218,47 +233,206 @@ func UnmarshalAccessListMember(data []byte, opts ...MarshalOption) (*accesslist.
 	return &member, nil
 }
 
-// IsAccessListOwner will return true if the user is an owner for the current list.
-func IsAccessListOwner(identity tlsca.Identity, accessList *accesslist.AccessList) error {
+// IsAccessListOwner will determine if the user is an owner for the current list.
+// If ownership is inherited (through the user having Membership of a list which is an Owner of the provided `accessList`),
+// the user must also meet the ownership requirements of the parent list.
+func IsAccessListOwner(ctx context.Context, members AccessListsAndMembersGetter, identity tlsca.Identity, accessList *accesslist.AccessList) (bool, error) {
 	// An opaque access denied error.
 	accessDenied := trace.AccessDenied("access denied")
+	// Is the Ownership inherited, or explicit?
+	inherited := false
```

Comment:
This is confusing. 

Function says that we are chakchecking if a user is IsAccessListOwner no matter if it is a top lever owner or owner by nested access list.